### PR TITLE
Upgrade to Check Spelling v0.0.21

### PR DESCRIPTION
**Pull request by jsoref**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
   - [ ] This will cover #89510
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

This PR upgrades check-spelling to [v0.0.21](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.21)
This adds a couple of features, including [GitHub job summaries](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Step-Summary) (which can be much longer than comments, but are out-of-line).

Probably the most appealing interesting new feature for users of this repository is that you should be able to [update `expect.txt` easily from Windows](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Update-expect-command-line).

As always, if you have problems, please let me know.

-----

## Before reenabling the workflow

I believe you will want to delete these branches:

* @KevinLaMS's [kevinla-node](https://github.com/microsoft/winget-pkgs/tree/kevinla-node)
* [preview-manifests-snapshot](https://github.com/microsoft/winget-pkgs/tree/preview-manifests-snapshot)

This branch should be rebased:
* @stephengillie's [stephengillie-patch-1](https://github.com/microsoft/winget-pkgs/tree/stephengillie-patch-1)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/90103)

### CODE REVIEW

File: .github/workflows/spellCheck.yaml [hunk 1]
```diff
@@ -26,29 +67,48 @@ jobs:
...
-        dictionary_source_prefixes: '{"cspell": "https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20220427/dictionaries/","cspell1": "https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20220814/dictionaries/"}'
+        use_magic_file: 1
+        extra_dictionary_limit: 10
         extra_dictionaries:
-          cspell:powershell/powershell.txt
-          cspell:typescript/typescript.txt
-          cspell:lorem-ipsum/dictionary.txt
-          cspell1:software-terms/src/software-terms.txt
+          cspell:software-terms/src/software-terms.txt
+          cspell:filetypes/filetypes.txt
```
Comment (jsoref) [inline .github/workflows/spellCheck.yaml:81]: When I refresh, I start from scratch and select things roughly in the order that the action suggests.

For reasons I'm not entirely certain, powershell / typescript / lorem were not highly ranked.

I suspect the reason lorem isn't highly ranked is that the lorem pattern already masked all the interesting lorem items. It's possible that the pattern could be removed and the dictionary used instead, but, it doesn't really matter.

File: .github/actions/spelling/patterns.txt [hunk 2]
```diff
@@ -24,11 +24,24 @@
 # FAQ items
 \`[A-Z]\`[a-z]{2,}\b
 
+# fabricbot
+"(?:body|comment)Pattern": ".*"
+
+# Moderators
+\*\*\s*\[@[^\]]+\]\([^)]+\)\*\*
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:31]: As noted in the PR, this covers another PR.

File: .github/actions/spelling/patterns.txt [hunk 3]
```diff
@@ -24,11 +24,24 @@
 # FAQ items
 \`[A-Z]\`[a-z]{2,}\b
 
+# fabricbot
+"(?:body|comment)Pattern": ".*"
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:28]: This line is responsible for most of the lines being removed from expect.txt.

**Review by Trenly — APPROVED** [review]

**Review by denelon — APPROVED** [review]

**Comment by jsoref** [comment]
@denelon oh, those are merely notes explaining how things work. I've resolved them to avoid confusing anyone

**Comment by jsoref** [comment]
@stephengillie: can you please delete and rebase the branches noted in the description above?

**Comment by Lut9310** [comment]
> - [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
> - [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
>    - [ ] This will cover #89510
> - [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
> - [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
> - [ ] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?
> 
> Note: `<path>` is the name of the directory containing the manifest you're submitting.
> 
> -----
> 
> This PR upgrades check-spelling to [v0.0.21](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.21)
> This adds a couple of features, including [GitHub job summaries](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Step-Summary) (which can be much longer than comments, but are out-of-line).
> 
> Probably the most appealing interesting new feature for users of this repository is that you should be able to [update `expect.txt` easily from Windows](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Update-expect-command-line).
> 
> As always, if you have problems, please let me know.
> 
> -----
> 
> ## Before reenabling the workflow
> 
> I believe you will want to delete these branches:
> 
> * @KevinLaMS's [kevinla-node](https://github.com/microsoft/winget-pkgs/tree/kevinla-node)
> * [preview-manifests-snapshot](https://github.com/microsoft/winget-pkgs/tree/preview-manifests-snapshot)
> 
> This branch should be rebased:
> * @stephengillie's [stephengillie-patch-1](https://github.com/microsoft/winget-pkgs/tree/stephengillie-patch-1)
> 
> 
> ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/90103)

**Pull request merged**